events/views.py

```diff
@@ -68,32 +70,33 @@ def event_register(request, id):
 
 
 # TODO: simplify with drf
-@api_view(['POST', ])
-@ecell_user
-def event_unregister(request, id):
-    u = request.ecelluser
-    if u:
-        try:
-            e = Event.objects.get(id=id)
-        except:
-            res_message="Event does not exist"
-            res_status=status.HTTP_404_NOT_FOUND   
-        else:
+class EventUnregisterView(APIView):
+    permission_classes = [IsAuthenticated]
+    @swagger_auto_schema(
+        operation_id="eventunregister",
+        responses={
+            '200' : set_example(responses.event_registration_deleted_200),
+            '403' : set_example(responses.user_forbidden_403),
+            '404' : set_example(responses.event_does_not_exist_404,responses.event_not_registered_404)
+        }
+    )
+    def post(request, id):
+        user = request.user
+        if user:
             try:
-                reg = EventRegister.objects.filter(user = u, event= e)  
+                event = Event.objects.get(id=id)
             except:
-                res_message= "Event not registered"
-                res_status=status.HTTP_404_NOT_FOUND
+                return Response(responses.event_does_not_exist_404,status.HTTP_404_NOT_FOUND) 
             else:
-                res_message="Registration deleted successfully"
-                reg.delete()
-                res_status=status.HTTP_200_OK
-    
-    else:
-        res_message = "Login to continue"
-    return Response({
-        "message": res_message
-    }, status=res_status)
+                try:
+                    reg = EventRegister.objects.filter(user = user, event= event)  
```

Comment:
`filter` returns the queryset satisfying a condition. If user isn't registered, it will return an empty queryset. Hence this isn't the appropriate way to check, as I'm getting the `"Registration deleted successfully"` in case of an empty queryset too.